src/app/perspective-status.component.ts

```diff
@@ -236,34 +236,45 @@ export class PerspectiveStatus implements OnChanges, AfterViewInit, AfterViewChe
    * callback, to ensure that the ViewChild has been updated.
    */
   ngAfterViewChecked() {
-    // TODO(rachelrosen): Queue up all animations from ngOnChanges and put them
-    // in a proper timeline (instead of just chaining promises), so state
-    // doesn't accidentally get messed up. This is not a production concern
-    // because state will not be changing there; it's only a concern for changing
-    // settings with the customizable form.
-    let updatesToDo = Promise.resolve();
-    if (this.scoreThresholdsChanged) {
-      this.scoreThresholdsChanged = false;
-      // Skip the updateWidgetStateAnimation if the loading style also changed,
-      // since the loading style update takes care of the widget state change
-      // animation.
-      if (!this.loadingIconStyleChanged) {
-        updatesToDo.then(() => {
-          this.updateDemoSettingsAnimation = this.getUpdateWidgetStateAnimation();
-          this.updateDemoSettingsAnimation.play();
-        });
-      }
-    }
+    if (this.scoreThresholdsChanged || this.loadingIconStyleChanged) {
+      let afterChangesTimeline = new TimelineMax({
+        onStart: () => {
+          this.ngZone.run(() => {
+            console.debug('Started animations in ngAfterViewChecked');
+          });
+        },
+        onComplete: () => {
+          this.ngZone.run(() => {
+            console.debug('Completing animations in ngAfterViewChecked');
+          });
+        }
+      });
+      // Run in a Promise resolve statement so we don't get an
+      // ExpressionChangedAfterItHasBeenCheckedError.
+      Promise.resolve().then(() => {
+        if (this.scoreThresholdsChanged) {
+          console.debug('Setting scoreThresholdsChanged to false');
+          this.scoreThresholdsChanged = false;
+          // Skip the updateWidgetStateAnimation if the loading style also changed,
```

Comment:
It looks to me like you could get rid of the comment and make the code clearer with something like
```
if (this.loadingIconStyleChanged || this.scoreThresholdsChanged) {
  // animation code
  if (this.loadingIconStyleChange) {
    // the specific stuff
  }
  if (this.scoreThresholdsChanged) {
    // other specific stuff
  }
}
```

I did notice that the animation is called slightly differently in the two cases though so that might work.  I think it would still be useful from an avoiding-surprises standpoint to stick with an `if` and `else if` for the animation instead of the custom-y control structure you have here.

Reply:
Thank you for the feedback. I was trying to avoid that kind of nesting but I guess it lost readability along the way! I will go back in and edit it to be clearer.